### PR TITLE
feat: preparation for v11 & gh api rate limiting

### PR DESCRIPTION
**Pull request by reggi**
Adds a release separation for v10 and v11 branches in npm.

* adds `gh` exec to get token so can be run locally
* adds caching for each version sha, and if match dont fetch files `cache.json`
* drops fetching v6 / v7 branches
* pulls in 11

A part of npm 11 roadmap: https://github.com/npm/statusboard/issues/898

### CODE REVIEW

File: cli/bin/build.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const checkContent = () => {
 build({
   releases: require('../releases.json'),
   loglevel: process.argv.includes('--debug') || process.env.CI ? 'verbose' : 'info',
-  prerelease: false,
+  prerelease: true,
```
Comment (wraithgar) [inline cli/bin/build.js:30]: Was this required cause 11 is in pre-release mode?

File: cache.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "v11": "cf52b8be2645ee098ee83ea9981fc32a11932fad",
```
Comment (wraithgar) [inline cache.json:2]: opinion: either the file should be named more specifically than "cache" or the entries should indicate they're cli repo shas.

File: cli/lib/build.js [hunk 3]
```diff
@@ -114,9 +114,11 @@ const main = async ({loglevel, releases: rawReleases, useCurrent, navPath, conte
   })
 
   const updates = await Promise.all(
-    releases.map(r => extractRelease(r, {contentPath, baseNav: navData, prerelease})),
+    releases.map(r => extractRelease(r, {cache, contentPath, baseNav: navData, prerelease})),
   ).then(r => r.filter(Boolean))
 
+  await cache?.save()
```
Comment (wraithgar) [inline cli/lib/build.js:120]: when is cache not present?

Comment (reggi) [reply cli/lib/build.js:120]: the tests dont tests the cache so it's an optional addition

File: cache.json (hunk 2 above)
Comment (reggi) [reply cache.json:2]: `npm-cli-latest-version-sha-cache-map.json`?

**Review by wraithgar — APPROVED** [review]

**Pull request merged**